### PR TITLE
Add OGRE 1.10.12 feedstock.

### PR DESCRIPTION
**Pull request by seanyen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by seanyen** [comment]
cc @wolfv

**Comment by wolfv** [comment]
Hi @seanyen this is awesome! 

My local recipe looked like this https://gist.github.com/wolfv/ccae7c22e2986502cc6dc2eef9005d8c
You can find some linux-things in there that i think are necessary.

Would you mind adding me in the mainainers section of this recipe?

Also I think adding run_exports here is appropriate so that other packages built against OGRE automatically have a runtime dependency on the OGRE library.

One other request: can we name the package `ogre3d`? Because that's what I've already put in the ros-conda-base-environment here: https://github.com/conda-forge/ros-conda-base-feedstock/blob/master/recipe/meta.yaml

**Comment by wolfv** [comment]
btw my recipe was just a prototype, it might not have the minimal set of dependencies etc., just wanted to post it as a reference (it worked for me as part of RViz like that).

**Comment by seanyen** [comment]
@wolfv I added you as maintainer there. For the naming, I would argue we should keep the name as consistent as possible to other package management using. https://index.ros.org/search/?term=ogre&section=deps

**Comment by seanyen** [comment]
@conda-forge/staged-recipes hi, is there any means I can enable Visual Studio 2017 for AppVeyor build? or how to skip it? I have Linux and Windows build verified on Azure DevOps.

File: recipes/ogre/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      -DOGRE_BUILD_COMPONENT_JAVA=FALSE \
+      -DOGRE_BUILD_SAMPLES=FALSE \
+      -DOGRE_BUILD_TESTS=FALSE \
+      -DOGRE_BUILD_DEPENDENCIES=FALSE \
+      -DOGRE_BUILD_TOOLS=FALSE \
+      -DOGRE_INSTALL_PDB=FALSE \
+      -DOGRE_RESOURCEMANAGER_STRICT=2 \
+      -DOGRE_THREAD_SUPPORT=2 \
+      -DOGRE_THREAD_PROVIDER=1
+make
```
Comment (chrisburr) [inline recipes/ogre/build.sh]:
```suggestion
make -j${CPU_COUNT}
```

File: recipes/ogre/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (chrisburr) [inline recipes/ogre/LICENSE]: This file isn't needed as conda-build will automatically get it from the source tarball instead.

**Comment by chrisburr** [comment]
I think you can ignore Appveyor as Azure is now the default for Windows builds in the feedstocks.

(I'm don't know enough about windows to know if the `conda_build_config.yml` file is set up the "right" way)

**Comment by seanyen** [comment]
> (I'm don't know enough about windows to know if the `conda_build_config.yml` file is set up the "right" way)

I shamelessly copied it from other feedstock, and my intent is to use Visual Studio 2017 for Windows build. If anyone read this comment and knows any better way, please let me know!

**Comment by seanyen** [comment]
@chrisburr Thanks for the feedback. I edited the pull request and let me know how to make it move forward!

**Comment by chrisburr** [comment]
Generally pinging the `@conda-forge/staged-recipes` team is the right thing to do to get things merged.

I don't think there is anything left outstanding so LGTM 👍 Thanks @seanyen!

**Comment by seanyen** [comment]
@conda-forge/staged-recipes Just wondering, after the staged recipe got merged, how long does it take to show up in the conda-forge channel for install? Or do I need to do anything additional to get it worked?

**Comment by wolfv** [comment]
There is a bug where CI isn't initially triggered automatically. :/ 
If you push an empty commit to master and CI runs on the feedstock, it should appear on anaconda.org soon and then propagate to CDNs within an hour so that its installable.

**Comment by jakirkham** [comment]
Yeah I retriggered this manually already. Should be coming through.

**Comment by jakirkham** [comment]
The issue @wolfv is referring to is this one ( https://github.com/conda-forge/conda-smithy/issues/1154 ).

**Pull request merged**